tests/conftest.py

```diff
@@ -43,39 +40,99 @@ def stub_command(self, cmdname, commands_map: Dict[str, str]):
              'AWS_ECR_USERNAME=username\nAWS_ECR_PASSWORD=password\n'})
         """
 
-        command_stub.generate(self.tmpfile(cmdname), commands_map)
+        command_stub.generate(str(self.tmp_file_path(cmdname)), commands_map)
 
-    def run(self, command: str):
-        """Runs command in the exec context"""
-        with open(self.tmpfile("main.sh"), "w") as main_script:
+    def prepare_run_script(self, command: str, target_tmp_dir=None) -> Path:
+        """Create a shell script for running command and return script path."""
+        script_name = "main.sh"
+        script_path = self.tmp_file_path(script_name)
+        with open(script_path, "w") as main_script:
             main_script.write("#!/bin/bash\n")
+            # write env vars
+
+            for env_name, env_value in self.environ.items():
+                # map any tmp_dir paths to the target_temp_dir, useful for docker mounts
+                tmp_dir_path = Path(self.tmp_dir)
+                if (
+                    target_tmp_dir
+                    and isinstance(env_value, Path)
+                    and tmp_dir_path in env_value.parents
+                ):
+                    tail = env_value.relative_to(tmp_dir_path)
+                    env_value = Path(target_tmp_dir) / tail
+                main_script.write(f"export {env_name}='{env_value}'\n")
+
+            # adjust curdir and PATH
+            # this ensures the stub commands get invoked and not any other commands available
+            # elsewhere on PATH
+            main_script.write('cd "$(dirname "$0")"\n')
+            main_script.write("export PATH=.:$PATH\n")
```

Comment:
Smart. I like doing this inside our generated script more than modifying the copied environment.

Reply:
Yeah I realized making the generated script do this as well as set the environment makes it more portable to docker.